stl/inc/ranges

```diff
@@ -7698,6 +7704,388 @@ namespace ranges {
+            static constexpr bool _Iterators_derive_from_tag =
+                (derived_from<
+                     typename iterator_traits<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>::iterator_category,
+                     _Tag_type>
+                    && ...);
+
+        public:
+            using iterator_category = decltype([]() noexcept {
+                using _Invoke_t = invoke_result_t<_Maybe_const<_IsConst, _Func>&,
+                    range_reference_t<_Maybe_const<_IsConst, _ViewTypes>>...>;
+
+                if constexpr (!is_reference_v<_Invoke_t>) {
+                    return input_iterator_tag{};
+                } else if constexpr (_Iterators_derive_from_tag<random_access_iterator_tag>) {
+                    return random_access_iterator_tag{};
+                } else if constexpr (_Iterators_derive_from_tag<bidirectional_iterator_tag>) {
+                    return bidirectional_iterator_tag{};
+                } else if constexpr (_Iterators_derive_from_tag<forward_iterator_tag>) {
+                    return forward_iterator_tag{};
+                } else {
+                    return input_iterator_tag{};
+                }
+            }());
+        };
+
+        template <bool _IsConst>
+        class _Iterator : public _Category_base<_IsConst> {
+        private:
+            friend zip_transform_view;
+
+            using _Parent_t = _Maybe_const<_IsConst, zip_transform_view>;
+            using _Base_t   = _Maybe_const<_IsConst, _Inner_view>;
+
+            _Parent_t* _Parent = nullptr;
+            /* [[no_unique_address]] */ _Ziperator<_IsConst> _Inner;
+
+            constexpr _Iterator(_Parent_t& _Parent_, _Ziperator<_IsConst> _Inner_) noexcept(
+                is_nothrow_move_constructible_v<_Ziperator<_IsConst>>) // strengthened
+                : _Parent(_STD addressof(_Parent_)), _Inner(_STD move(_Inner_)) {}
+
+        public:
+            using iterator_concept = typename _Ziperator<_IsConst>::iterator_concept;
+            using value_type       = remove_cvref_t<invoke_result_t<_Maybe_const<_IsConst, _Func>&,
+                range_reference_t<_Maybe_const<_IsConst, _ViewTypes>>...>>;
+            using difference_type  = range_difference_t<_Base_t>;
+
+            _Iterator() = default;
+
+            constexpr _Iterator(_Iterator<!_IsConst> _Rhs) noexcept(
+                is_nothrow_convertible_v<_Ziperator<false>, _Ziperator<_IsConst>>) // strengthened
+                requires (_IsConst && convertible_to<_Ziperator<false>, _Ziperator<_IsConst>>)
+                : _Parent(_Rhs._Parent), _Inner(_STD move(_Rhs._Inner)) {}
+
+            _NODISCARD constexpr decltype(auto) operator*() const
+                noexcept(noexcept(_STD apply(_Dereference_closure(), _Inner._Current))) {
+                return _STD apply(_Dereference_closure(), _Inner._Current);
+            }
+
+            constexpr _Iterator& operator++() noexcept(noexcept(++_Inner)) // strengthened
+            {

```

Comment:
I noticed that there are several occurrences of `// strengthened` forcing the next `{` to wrap to a new line. We usually use `/* strengthened */` to avoid this (one of the few times we use C-style comments).